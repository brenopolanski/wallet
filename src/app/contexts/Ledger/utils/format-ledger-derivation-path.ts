import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
