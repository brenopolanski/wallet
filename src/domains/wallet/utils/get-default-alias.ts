import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
