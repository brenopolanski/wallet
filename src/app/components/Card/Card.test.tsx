import userEvent from "@testing-library/user-event";
import React from "react";
import { fireEvent, render } from "utils/testing-library";

import { Card } from "./Card";

describe("Card", () => {
	it("should render", () => {
		const { container, asFragment } = render(<Card />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should handle click", () => {
		const handleClick = jest.fn();
		const { container, asFragment, getByText } = render(<Card onClick={() => handleClick()}>Test</Card>);

		expect(container).toBeInTheDocument();

		userEvent.click(getByText("Test"));

		expect(handleClick).toHaveBeenCalledWith();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should selected", () => {
		const { container, asFragment } = render(<Card isSelected={true} />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
