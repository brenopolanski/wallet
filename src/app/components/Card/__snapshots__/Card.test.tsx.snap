// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should handle click 1`] = `
<DocumentFragment>
  <div>
    <button
      class="Cardstyles__CardButton-sc-ajtauu-0 dZiOWL"
      data-testid="Card"
      type="button"
    >
      Test
      <div
        class="flex absolute -top-1 -right-1 items-center m-4 space-x-1"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should render 1`] = `
<DocumentFragment>
  <div>
    <button
      class="Cardstyles__CardButton-sc-ajtauu-0 gawQiQ"
      data-testid="Card"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex absolute -top-1 -right-1 items-center m-4 space-x-1"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Card should selected 1`] = `
<DocumentFragment>
  <div>
    <button
      class="Cardstyles__CardButton-sc-ajtauu-0 gCeoSb"
      data-testid="Card"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex absolute -top-1 -right-1 items-center m-4 space-x-1"
      />
    </button>
  </div>
</DocumentFragment>
`;
